etna/blog/models.py

```diff
@@ -73,6 +74,19 @@ class BlogPostPage(
         BlogPostPageStreamBlock(),
     )
 
+    @cached_property
+    def type_label(cls) -> str:
+        """
+        Overrides the type_label method from BasePage, to return the correct
+        type label for the blog post page.
+        """
+        if parent := cls.get_parent():
+            print(parent.get_parent().specific_class)
+            if parent.get_parent().specific_class == BlogPage:
+                return parent.get_parent().title
+            return parent.title
+        return "Blog post"
```

Comment:
Could this be an option instead? This would then work for any level, right?

```suggestion
        parent = cls.get_parent()
        if not parent:
            return "Blog post"
        while parent.get_parent().specific_class == BlogPage:
            parent = parent.get_parent()
        return parent.get_parent().title
```